x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/AnalyticsCollectionService.java

```diff
@@ -74,20 +81,33 @@ public void putAnalyticsCollection(
         // This operation is supposed to be executed on the master node only.
         assert (state.nodes().isLocalNodeElectedMaster());
 
-        try {
-            analyticsCollectionResolver.collection(state, request.getName());
-            listener.onFailure(new ResourceAlreadyExistsException("analytics collection {} already exists", request.getName()));
-        } catch (ResourceNotFoundException e) {
-            AnalyticsCollection collection = new AnalyticsCollection(request.getName());
-            CreateDataStreamAction.Request createDataStreamRequest = new CreateDataStreamAction.Request(collection.getEventDataStream());
+        AnalyticsCollection collection = new AnalyticsCollection(request.getName());
+        CreateDataStreamAction.Request createDataStreamRequest = new CreateDataStreamAction.Request(collection.getEventDataStream());
+
+        ActionListener<AcknowledgedResponse> createDataStreamListener = ActionListener.wrap(
+            r -> listener.onResponse(new PutAnalyticsCollectionAction.Response(r.isAcknowledged(), request.getName())),
+            (Exception e) -> {
+                if (e instanceof ResourceAlreadyExistsException) {
+                    listener.onFailure(
+                        new ResourceAlreadyExistsException("analytics collection [{}] already exists", request.getName(), e)
+                    );
+                    return;
+                }
+
+                RestStatus status = RestStatus.INTERNAL_SERVER_ERROR;
+
+                if (e instanceof ElasticsearchException) {
+                    status = ((ElasticsearchException) e).status();
+                }
+
+                e = new ElasticsearchStatusException("error while creating analytics collection [{}]", status, e, request.getName());
+                logger.error(e.getMessage(), e);
```

Comment:
Do we need to log the errors? Up until now, we haven't done that in the module...